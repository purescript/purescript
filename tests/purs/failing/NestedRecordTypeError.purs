-- @shouldFailWith TypesDoNotUnify
module A where

-- Some labels are missing, these should be pointed out in the error 
f :: { a :: { c :: { d :: Int } }, d :: Int, e :: Int, f :: Int } -> Int
f { a: { b: { z } }, d, e, f, g } = z
