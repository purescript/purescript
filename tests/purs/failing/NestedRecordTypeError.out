Error found:
in module [33mA[0m
at tests/purs/failing/NestedRecordTypeError.purs:6:3 - 6:34 (line 6, column 3 - line 6, column 34)

  Could not match type
  [33m                          [0m
  [33m  ( a :: { c :: { d :: Int[0m
  [33m                }         [0m
  [33m         }                [0m
  [33m  ...                     [0m
  [33m  )                       [0m
  [33m                          [0m
  with type
  [33m                         [0m
  [33m  ( a :: { b :: { z :: t2[0m
  [33m                | t3     [0m
  [33m                }        [0m
  [33m         | t4            [0m
  [33m         }               [0m
  [33m  , g :: t1              [0m
  [33m  ...                    [0m
  [33m  | t5                   [0m
  [33m  )                      [0m
  [33m                         [0m

while trying to match type [33m{ a :: { c :: { d :: Int[0m
                           [33m              }         [0m
                           [33m       }                [0m
                           [33m, d :: Int              [0m
                           [33m, e :: Int              [0m
                           [33m, f :: Int              [0m
                           [33m}                       [0m
  with type [33mRecord t0[0m
while checking that expression [33mcase $0 of                                           [0m
                               [33m  { a: { b: { z: z } }, d: d, e: e, f: f, g: g } -> z[0m
  has type [33mInt[0m
in value declaration [33mf[0m

where [33mt0[0m is an unknown type
      [33mt5[0m is an unknown type
      [33mt4[0m is an unknown type
      [33mt3[0m is an unknown type
      [33mt2[0m is an unknown type
      [33mt1[0m is an unknown type

See https://github.com/purescript/documentation/blob/master/errors/TypesDoNotUnify.md for more information,
or to contribute content related to this error.

